primitives/xcm-commons/src/lib.rs

```diff
@@ -77,3 +77,22 @@ where
         matches!((asset.id.0.parents, asset.id.0.first_interior()), (1, Some(GlobalConsensus(network))) if *network == EthereumNetwork::get())
     }
 }
+
+/// Filter to ensure an ETH asset is coming from a Parachain.
+pub struct EthereumAssetReserveFromPara<EthereumLocation, EthereumNetwork>(
```

Comment:
We need to merge it to the single file as soon as another similar PR is merged: https://github.com/moondance-labs/tanssi/pull/1266/files#diff-2292d9b5199ef311280b1081176611a29b920776ab17127f8de3d81eb0c8bcfdR221